### PR TITLE
[trivial] Separate insert from insert_final_value in versioned_aggregators

### PR DESCRIPTION
**Pull request by igor-aptos**
to have better enforcement (and fix the current issue)

### CODE REVIEW

File: aptos-move/mvhashmap/src/versioned_delayed_fields.rs [hunk 1]
```diff
@@ -155,20 +155,12 @@ impl<K: Copy + Clone + Debug + Eq> VersionedValue<K> {
...
-                    // returning 'true' ensures that the bypass enabling logic is not affected.
-                    (Value(val_l, None), Value(val_r, _)) if val_l == val_r => true,
-                    (Value(_, None), Apply(_)) => true,
-
                     // TODO: change to code_invariant_error
-                    (_, _) => unreachable!(
+                    (cur, new) => unreachable!(
                         "Replaced entry must be an Estimate, \
-			 or we should be recording the final committed value"
+			 or we should be recording the final committed value, {:?} to {:?}",
```
Comment (gelash) [inline aptos-move/mvhashmap/src/versioned_delayed_fields.rs:161]: I guess the comment needs to change here (no longer recording committed value)

File: aptos-move/mvhashmap/src/versioned_delayed_fields.rs [hunk 2]
```diff
@@ -182,6 +174,12 @@ impl<K: Copy + Clone + Debug + Eq> VersionedValue<K> {
         }
     }
 
+    fn insert_final_value(&mut self, txn_idx: TxnIndex, value: DelayedFieldValue) {
+        if let None = self.versioned_map.insert(txn_idx, CachePadded::new(VersionEntry::Value(value, None))) {
```
Comment (gelash) [inline aptos-move/mvhashmap/src/versioned_delayed_fields.rs:178]: Let's still assert (I guess we will change to CodeError) more than not None, it should be a delta (or maybe a value with the same value). Or maybe that it's not an estimate.

**Review by gelash — APPROVED** [review]

**Review by georgemitenkov — APPROVED** [review]

File: aptos-move/block-executor/src/executor.rs [hunk 3]
```diff
@@ -153,9 +153,7 @@ where
             }
 
             for (id, change) in output.delayed_field_change_set().into_iter() {
-                // if !prev_modified_aggregators.remove(&id) {
-                //     updates_outside = true;
-                // }
+                prev_modified_delayed_fields.remove(&id);
```
Comment (vusirikala) [inline aptos-move/block-executor/src/executor.rs:156]: Earlier, the `updates_outside` flag was set to true when a new aggregator is changed during this incarnation. But now, we are not setting updates_outside flag. What's the rationale?

Comment (georgemitenkov) [reply aptos-move/block-executor/src/executor.rs:156]: I think there is also a todo below for that

Comment (igor-aptos) [reply aptos-move/block-executor/src/executor.rs:156]: yes there is TODO - but I've discussed with @gelash - I don't think we want to update it. It marks later transactions for re-validation - but we don't look at delayed fields during re-valiadation - only before commit. (at the moment)

**Pull request merged**